### PR TITLE
edit Dispatch license info so that GitHub recognizes it

### PR DESCRIPTION
**Pull request by eirinikos**
Hello! 👋

(CCing @dankohn who has requested this work so that the license will appear correctly in https://landscape.cncf.io/selected=dispatch)

GitHub uses a library called Licensee to identify a project's license
type. It shows this information in the status bar and via the API if it
can unambiguously identify the license.

This commit updates the LICENSE file so that it contains only the full
text of the Apache license and references VMWare, Inc as the copyright
holder. The info that pertains to 3rd-party software licensing has now
been moved to a new DEPENDENCIES file. Finally, the README file has a
new "Licensing" section to explain to readers the purposes of the
LICENSE and DEPENDENCIES files.

Collectively, these changes allow Licensee to successfully identify the
license type of Dispatch as Apache.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,3 +50,16 @@ $ dispatch install --file config.yaml
...
 
 For a more complete quickstart see the [developer documentation](#documentation)
+
+## Licensing
+
+Dispatch is licensed under the [Apache 2 license](./LICENSE.txt).
+
+Dispatch version 0.1.0 includes a number of components with separate copyright notices and license terms. Please see [DEPENDENCIES.txt](./DEPENDENCIES.txt) for more information.
+
+=============== TABLE OF CONTENTS =============================
```
Comment (tenczar) [inline README.md]: Is this Table of Contents meant to be here?

Comment (eirinikos) [reply README.md]: oh, good catch! Sorry for overlooking this - I'll push a new commit to remove the TOC reference.

**Comment by eirinikos** [comment]
Hey @tenczar, thanks for letting me know. And thanks for pointing out NOTICE.txt - I'd overlooked this file when I made my changes.

AFAIK, there's no need to update this before Monday. In the meantime, I'll be on standby in case you need anything from my end.

**Comment by tenczar** [comment]
Hey Andrea, we've received updated files directly from our open source legal team. Those files have now been merged into the source tree. Do these changes support licensee and meet the requirements for cncf? We can reopen this PR if changes are still required. Thanks.

**Comment by eirinikos** [comment]
Hey @tenczar, thanks for circling back. It looks like Licensee still doesn't detect the license properly. (At the bottom of this comment is the output that I get when I run `licensee detect` on Dispatch's remote repo.)

I do have some ideas for other changes that we can try. Could we reopen this PR? Thanks!

```
$ licensee detect https://github.com/vmware/dispatch
License:        Other
Matched files:  LICENSE.txt, open_source_license_dispatch_2018.txt
LICENSE.txt:
  Content hash:  fe1eb7c04c48176d711013903085b2fd6b999a17
  License:       Other
  Closest licenses:
    Apache-2.0 similarity:  98.80%
    ECL-2.0 similarity:     95.92%
    EPL-2.0 similarity:     57.48%
open_source_license_dispatch_2018.txt:
  Content hash:  fe1eb7c04c48176d711013903085b2fd6b999a17
  License:       Other
  Closest licenses:
    Apache-2.0 similarity:  98.80%
    ECL-2.0 similarity:     95.92%
    EPL-2.0 similarity:     57.48%
```

**Comment by tenczar** [comment]
Sure, let's get this working.

**Comment by kars7e** [comment]
Looks like @markpeek already fixed it via https://github.com/vmware/dispatch/pull/406. I see license being properly detected when visiting https://github.com/vmware/dispatch/blob/47ef1bca6497e2984b73f30ffe60b179fc7bb507/LICENSE. It seems still not updated on https://landscape.cncf.io/selected=dispatch, @eirinikos can you take a look?

**Comment by kars7e** [comment]
Looks like https://landscape.cncf.io/selected=dispatch now properly recognizes the license! Closing :-)

**Pull request closed without merging**